book.rb

```diff
@@ -0,0 +1,17 @@
+require_relative 'item'
+
+class Book < Item
+  attr_accessor :publisher, :cover_state
+
+  def initialize(publisher, cover_state, publish_date, id: Random.rand(1..1000), archived: false)
+    super(publish_date, id: id, archived: archived)
+    @publisher = publisher
+    @cover_state = cover_state
+  end
+
+  private
+
+  def can_be_archived?
+    super || @cover_state == 'bad'
```

Comment:
[Optional]
It's a great job you overridden the implementation of `can_be_archived?`. But what if the user inputs the status as `Bad or BAD`? Thus, I suggest you consider this case. You can use case conversion like `@cover_state.downcase` .